### ISSUE TITLE
Yolov8 performing poorly on object detection task involving detecting cars and polythene trash.

### ISSUE THREAD

**Issue by Takshil12345** [ultralytics/ultralytics#8419]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I tried to make a litter detection model which detects litter (specifically polythene bags and paper bags) thrown out of moving cars with the idea of issuing fines against those cars . The idea is inspired from https://www.youtube.com/watch?v=M1zf8Cp4AS8&list=PLL5bhsIuEBbCy6FbI2Ax2HiZCB1I3bnxl&index=2 while i am trying to implement it specifially for polythene and paper bags.
For this I took separate datasets of vehicles  and polythen&paper trash and combined them into one dataset while generating their labels. I used yolov8x (pre trained) object detection model and trained it for 24 epochs. The train folder contains 47k images of which 50% of the images belong to vehicles class and the remaining 50% to trash class while the valid folder contains 11k images of which 55% are vehicles and 45% trash.

The problem is that the class( vehicle) having more instances has less mAP and the class Trash having less instances has more mAP but when i test my model on random images, the model predicts vehicles very accurately but fails to detect any trash in the image. How is this inconsistency possible ? Should the model be doing the exact opposite speaking from a theoretical point of view.

<img width="1010" alt="Screenshot 2024-02-23 at 2 05 01 AM" src="https://github.com/ultralytics/ultralytics/assets/92371620/1d3d88f2-d8b1-403d-8b91-012345a3dbfe">

As visible in the above image the mAP value for trash is very high but the model fails to predict trash correctly on the images i tested . 

Any help would be greatly appreciated.

**Comment by Burhan-Q** [comment]
@Takshil12345 I commented on your post in the Discord server as well, but will follow up with you here.

The problem is more likely that `trash` is too broad of a classification. There are lots of different types of objects and they won't have similar enough appearance to be classified well, where as a road vehicle (cars, trucks, etc) generally have a common structure.

Your training results are likely misleading because you only trained for only 24 epochs. What did the train/val loss plots look like? I expect you'll need to train for much longer.

**Comment by Takshil12345** [comment]
Hi @Burhan-Q thanks alot for following up , coming to trash being too broad to classify i do completely agree with this and thats why i decided to specifically go with polythene and paper bags as trash only as i felt that this would help reduce the amout of variety in trash.

![polythene bag](https://github.com/ultralytics/ultralytics/assets/92371620/4e4bf453-f5b6-49ab-8a02-a14084debe07)
![disposable-bag](https://github.com/ultralytics/ultralytics/assets/92371620/66761f02-5282-41d3-a59c-1613064cc2ff)

I have only included these 2 types of trash in my dataset and have labeled them under the category of trash with the idea of reducing the amout of variety so that the model would be able to grab important features from it. i took help of the data set give here : https://universe.roboflow.com/firstworkspace-jg76u/bags-41emo

I used the yolov8x model for this but since it was taking around 10-11 hours to train just 6 epochs had to do this 4 times thus my validation and training graphs don't exactly represent 24 epochs in one go , but apart from the the losses have been decreasing as such for the most part.

How many epochs would you recommend running this model for and dont you feel seeing the results from the validation set output that the model would eventually overfit on trash since the trash mAP is already very high ?

**Comment by Takshil12345** [comment]
Hi @Burhan-Q thanks alot for such a detailed response will surely take into consideration these insights from here on , just wanted to ask one more thing 
https://arxiv.org/pdf/2401.14719.pdf : This is a paper where the researchers made a dataset called pLitter which consists of road side images of trash , although it does have a lot of trash categories but do you think that using something like this would translate to better results when i am trying to detect trash being thrown from vehicles on the road , say i want to achieve something similar to this : https://www.youtube.com/watch?v=M1zf8Cp4AS8&list=PLL5bhsIuEBbCy6FbI2Ax2HiZCB1I3 then do you think that using the images of litter from pLitter like the one below would help ? 
![Screenshot 2024-02-26 at 12 10 12 PM](https://github.com/ultralytics/ultralytics/assets/92371620/eb7e73d2-3a15-4caf-8e81-ea975252b19a)

Would images taken via the above configuration perform any better than the one below which has been made synthetically in your opinion since i dont have access to any such videos of trash being thrown and am dependent on images of trash from from different datasets to make my neural network? 
![Screenshot 2024-02-26 at 12 12 41 PM](https://github.com/ultralytics/ultralytics/assets/92371620/a4746acc-b021-41e2-b12c-795d7183693d)
